frontend/src/app/shared/components/fields/display/field-types/compound-progress-display-field.module.ts

```diff
@@ -0,0 +1,140 @@
+import { DisplayField } from 'core-app/shared/components/fields/display/display-field.module';
+import { InjectField } from 'core-app/shared/helpers/angular/inject-field.decorator';
+import { PathHelperService } from 'core-app/core/path-helper/path-helper.service';
+import { ProjectResource } from 'core-app/features/hal/resources/project-resource';
+import { uiStateLinkClass } from 'core-app/features/work-packages/components/wp-fast-table/builders/ui-state-link-builder';
+import * as URI from 'urijs';
+
+export class CompoundProgressDisplayField extends DisplayField {
+  @InjectField() PathHelper:PathHelperService;
+
+  @InjectField() apiV3Service:ApiV3Service;
+
+  private derivedText = this.I18n.t('js.label_value_derived_from_children');
+
+  public render(element:HTMLElement, displayText:string):void {
+    element.classList.add('split-time-field');
+    element.setAttribute('title', displayText);
+
+    this.renderActual(element, displayText);
+
+    if (this.hasChildren()) {
+      this.renderSeparator(element);
+      this.renderDerived(element, this.derivedValueString);
+    }
+  }
+
+  private renderActual(element:HTMLElement, displayText:string):void {
+    const span = document.createElement('span');
+
+    span.textContent = displayText;
+    span.title = this.valueString;
+    span.classList.add('-actual-value');
+
+    element.appendChild(span);
+  }
+
+  private renderDerived(element:HTMLElement, displayText:string):void {
+    const link = document.createElement('a');
+
+    link.textContent = `Σ ${displayText}`;
+    link.title = `${displayText} ${this.derivedText}`;
+    link.classList.add('-derived-value', uiStateLinkClass);
+
+    this.addURLToViewWorkPackageChildren(link);
+
+    element.appendChild(link);
+  }
+
+  public renderSeparator(element:HTMLElement):void {
+    const span = document.createElement('span');
+
+    span.textContent = '·';
+    span.classList.add('-separator');
+    span.ariaHidden = 'true';
+
+    element.appendChild(span);
+  }
+
+  private addURLToViewWorkPackageChildren(link:HTMLAnchorElement):void {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    if (this.resource && this.resource.id && this.resource.project) {
+      const wpID = this.resource.id.toString();
+      this
+        .apiV3Service
+        .projects
+        .id(this.resource.project as ProjectResource)
+        .get()
+        .subscribe((project:ProjectResource) => {
+          const props = {
+            c: ['id', 'subject', 'type', 'status', 'estimatedTime', 'remainingTime', 'percentageDone'],
+            hi: true,
+            is: true,
+            f: [{ n: 'parent', o: '=', v: [wpID] }],
+          };
+          const href = URI(this.PathHelper.projectWorkPackagesPath(project.identifier as string))
+            .query({ query_props: JSON.stringify(props) })
+            .toString();
+
+          link.href = href;
+        });
+    }
+  }
```

Comment:
The links to view work estimates of work package and its children should show the same query regardless of clinking on the derived work, remaining work or % complete.

The simplest to do so would be to have the code for `addURLToViewWorkPackageChildren` in `EstimatedTimeDisplayField` and `CompoundProgressDisplayField` deduplicated.

Reply:
Okay, I interpreted it as clicking on work related links leads to a work-focused query (sans % complete), and clicking on the % complete link leads to a query including the % complete column. Thanks for clarifying